pkg/kubelet/network/kubenet/kubenet_linux.go

```diff
@@ -513,13 +517,23 @@ func (plugin *kubenetNetworkPlugin) getRunningPods() ([]*hostport.RunningPod, er
 			continue
 		}
 		if pod, ok := plugin.host.GetPodByName(p.Namespace, p.Name); ok {
-			runningPods = append(runningPods, &hostport.RunningPod{
+			activePods = append(activePods, &hostport.ActivePod{
 				Pod: pod,
 				IP:  podIP,
 			})
 		}
 	}
-	return runningPods, nil
+	return activePods, nil
+}
+
+// podIsExited returns true if the pod is exited (all containers inside are exited).
```

Comment:
@freehan Thanks for catching. Update the code, also tests the pod sanboxes (which contains the infra container).
Only if all containers exites, it returns true.